### PR TITLE
Ensured restaurant website link has protocol (http, https)

### LINKED ISSUE #6: When adding a restaurant website, if 'http://' or 'https://' is not entered, URL will start `localeatsottawa.com/` instead

### PR DESCRIPTION
**Pull request by pascallaliberte**
Fixes #6 

- do model-based getter check to not burden contributor with validation message
- default to http://

### CODE REVIEW

**Comment by caseyli** [comment]
Hey @pascallaliberte thanks for contributing! I apologize for these issues not being described fully and properly as we're still rapidly developing this.

I like what you've done in terms of not burdening the user, however, I think I would like to see what you've done saved to the DB rather than overriding the accessor.

So would you be able to repurpose your code into a `before_save` callback on the model to add the http:// if the website doesn't start with it.

Thanks! Again huge appreciation on the contribution.

File: app/models/restaurant.rb [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
+  
   has_many :locations, dependent: :destroy
 
   validates :name, presence: true, uniqueness: true
 
   scope :alphabetical, -> { order(name: :asc)}
+  
+  private
+  
+  def ensure_website_has_protocol
```
Comment (caseyli) [inline app/models/restaurant.rb]: @pascallaliberte looks great! - can indent the private methods one level

```
class Restaurant < ApplicationRecord
  ...
  private
    def ensure_website_has_protocol
      ...
    end
end
```

File: db/migrate/20200322235224_ensure_website_urls_include_protocol.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class EnsureWebsiteUrlsIncludeProtocol < ActiveRecord::Migration[6.0]
+  def change
+      Restaurant.all.each do |restaurant|
```
Comment (caseyli) [inline db/migrate/20200322235224_ensure_website_urls_include_protocol.rb]: @pascallaliberte looks great! Just outdent once (looks like 2 spaces to far in)

**Comment by pascallaliberte** [comment]
Location also has a website field (unused for now?) and I looked into DRYing that sanitizer method (in a concern?).

Was looking at this guy's proposal, but looked monkey-pathy to me... 
https://stackoverflow.com/a/22202651/2414273

**Pull request merged**